### ISSUE TITLE
[10Rc1] issue installing with PHP 8.1

### ISSUE THREAD

**Issue by Giudy** [glpi-project/glpi#10696]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0 RC1

### Bug description

Having Php 8.1 it is impossible to install glpi as when the DB is created we get this error:

`Uncaught Exception mysqli_sql_exception: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ''0', `type` int(11) NOT NULL DEFAULT '0' COMMENT '...' at line 4 in /data/www/html/glpi/inc/dbmysql.class.php at line 328`

![immagine](https://user-images.githubusercontent.com/7987648/154523652-5dcd0dac-7a23-42f3-9da6-782bc96ac323.png)

The issue is related to a new default with php 8.1 that we have solved doing this change that should be managed through the different versions of php

Since php 8.1 it is needed to use the ENT_COMPAT flag.

```
--- inc/dbmysql.class.php.bug   2022-02-04 13:51:09.429373918 +0000 
+++ inc/dbmysql.class.php       2022-02-04 11:08:56.868996529 +0000 
@@ -853,7 +853,7 @@ 
      foreach ($queries as $query) { 
         $query = trim($query); 
         if ($query != '') { 
-            $query = htmlentities($query); 
+            $query = htmlentities($query, ENT_COMPAT); 
            if (!$this->query($query)) { 
               return false; 
```

**Comment by cedric-anne** [comment]
Hi,

This fix il already present in GLPI 10.0, see https://github.com/glpi-project/glpi/blob/26a36ecfbe427ffaaa5fc666f9d8e68e04e173b7/src/DBmysql.php#L974

`inc/dbmysql.class.php` file does not exist anymore (class files has been moved and renamed to fit PSR-4 convention), so I guess you are using a 9.5.x version of GLPI, which is not compatible with PHP 8.1.

I close this issue as PHP 8.1 compatibility of GLPI 9.5.x will not be implemented. Indeed, it requires too much work and GLPI 10.0.0 will be released soon.

Regards

**Comment by mzaourou** [comment]
Uncaught Exception mysqli_sql_exception: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ''0', `type` int(11) NOT NULL DEFAULT '0' COMMENT '...' at line 4 in C:\xampp\htdocs\glpi\inc\dbmysql.class.php at line 328
